### DISCUSSION TITLE
Do I have access to a simular app.listen in blitz?

### DISCUSSION THREAD

**Discussion by imcodingideas** [blitz-js/blitz#3035] [Q&A] [answered]
I am looking at two middlewares using uppy looking at this example:

https://github.com/transloadit/uppy/blob/main/examples/uppy-with-companion/server/index.js#L56
https://github.com/transloadit/uppy/blob/main/examples/uppy-with-companion/server/index.js#L69

Taking a look at `app.use(companion.app(uppyOptions))` I can use as a blitz middleware however, I cannot find a way to add app.listen `companion.socket(app.listen(3020))` as a middleware without a custom server. Am I missing something?

How can I go about filling these properties?
https://github.com/transloadit/uppy/blob/main/examples/uppy-with-companion/server/index.js#L47-L50

**Comment by beerose** [comment, marked as answer]
Unfortunately, the only way to have `app.listen` etc, is by using a custom server. Otherwise, you'd need to find a way to integrate these middlewares without it.

**Reply by imcodingideas** [reply]
Thanks!
